gnosis/cowswap
# Redirect hacked users to page

# Summary

This PR handle the issue with AnySwap hack to protect users, read more in:
https://cointelegraph.com/news/multichain-asks-users-to-revoke-approvals-amid-critical-vulnerability

The affected users will be redirected to:
![image](https://user-images.githubusercontent.com/2352112/150616658-dd784b7a-11a8-441a-b1d7-a1b489dd2daa.png)

It uses a hook that will decide if the user is affected or not.
It will check for WETH (main issue), and for WBNB, PERI and MATIC (could be affected too). I didn't find AVAX 🤷‍♀️

To glue this together, there's a component `RedirectAnySwapAffectedUsers` that will use the hook and redirect the user to the new page.

# To Test

1. Test that nothing changes in the app when you connect to mainnet or any other wallet
2. Uncomment line https://github.com/gnosis/cowswap/blob/205a59dbaddab1d67716c6eb0754da03600e29f9/src/custom/pages/error/AnySwapAffectedUsers/useIsAnySwapAffectedUser.ts#L23 to use the `VaultRelayer` instead of `AnySwapV4Router` contract to check your allowance. Since you probably have an allowance with the token WETH in mainnet, then you should see the error page.

# Not included
Please, ignore that the CoW image has a 404 on it. I think we made already too much effort to address this issue and should be good enough :)
